internal/value/value.go

```diff
@@ -2097,21 +2096,58 @@ func TextValue(v string) textValue {
 	return textValue(v)
 }
 
+type UUIDIssue1501FixedBytesWrapper struct {
+	val [16]byte
+}
+
+func NewUUIDIssue1501FixedBytesWrapper(val [16]byte) UUIDIssue1501FixedBytesWrapper {
+	return UUIDIssue1501FixedBytesWrapper{val: val}
+}
+
+// PublicRevertReorderForIssue1501 needs for fix uuid when it was good stored in DB,
+// but read as reordered. It may happen within migration period.
+func (w UUIDIssue1501FixedBytesWrapper) PublicRevertReorderForIssue1501() uuid.UUID {
+	return uuid.UUID(uuidFixBytesOrder(w.val))
+}
+
+func (w UUIDIssue1501FixedBytesWrapper) AsBytesArray() [16]byte {
+	return w.val
+}
+
+func (w UUIDIssue1501FixedBytesWrapper) AsBytesSlice() []byte {
+	return w.val[:]
+}
+
+func (w UUIDIssue1501FixedBytesWrapper) AsBrokenString() string {
+	return string(w.val[:])
+}
+
 type uuidValue struct {
-	value [16]byte
+	value               uuid.UUID
+	reproduceStorageBug bool
```

Comment:
It is not necessary a data corruption bug, maybe rename it to something like wrongBytesOrder